torch/csrc/jit/python/init.cpp

```diff
@@ -737,18 +737,10 @@ void initJITBindings(PyObject* module) {
     });
   });
 
-  py::class_<PythonFutureWrapper>(m, "Future")
+  py::class_<PythonFutureWrapper, std::shared_ptr<PythonFutureWrapper>>(m, "Future")
```

Comment:
why it becomes a shared_ptr binding?

Reply:
I wanted to kill `PythonFutureWrapper` copy constructor. 

And noticed these 2 sites will deep copy `PythonFutureWrapper`, because Python holds a unique_ptr underlyingly.
```
return obj.cast<std::shared_ptr<PythonFutureWrapper>>()->fut;
```

```
return py::cast(std::make_shared<PythonFutureWrapper>(ivalue.toFuture()));
```

And I remember the practice we use in "rpc/init.cpp" pybind bindings is that we always pass `shared_ptr` to our custom types to Python land. 
https://github.com/pytorch/pytorch/blob/a2557970f3eefe4c4ec3ff30b7d1c33d8b7bc246/torch/csrc/distributed/rpc/init.cpp#L29

It's not a big deal for this thin wrapper, but I want to switch to this better practice from now. 
Agree?

Reply:
It's a by-the-way thing, since I am touching surrounding code. 
I believe this is a better pattern. If you disagree, I can always split the PR.
My purpose is to avoid deep copy on PythonFutureWrapper. I know it's not important. (I noticed JIT has a few bindings that might incur unnecessary deep-copies.)
Since this Future wrapper will be used by RPC, which incurs millions of calls, I suggested we use the better practice from the get-go.

Because of possible later reference on the PythonFutureWrapper, pybind has to keep on holding the unique_ptr. When it's passed to C++ world, specifically, calling `obj.cast<PythonFutureWrapper>`, notice we were not casting to a pointer, so it must copy the object held by the unique_ptr.